README.md

```diff
@@ -493,6 +493,12 @@ name,iterations,real_time,cpu_time,bytes_per_second,items_per_second,label
 "BM_SetInsert/1024/10",106365,17238.4,8421.53,4.74973e+06,1.18743e+06,
 ```
 
+## Output Files
+The library supports writing the output of the benchmark to a file specified
+by `--benchmark_output=<filename>`. The format of the output can be specified
+using `--benchmark_output_format={json|console|csv}`. Specifying
+`--benchmark_output` does not suppress the console output.
```

Comment:
> how often do you think someone would want to push the output to tooling and have it human readable? 

I do it all the time. I want to see the benchmark results as the come in, and I also want to do something with the results later.

> , i think if you're going to push the output to a tool that tool should take care of the human readable bit too. 

I think that tool should take care of the human readable input for the information it presents, I don't expect the tool to repeat the benchmark output. 

> ie, i'd expect $ benchmark_my_stuff --benchmark_format=json | analyse_my_bench
> to provide me with human readable analysis results. if i need the output for debugging, then > tee or some equivalent would at least give me something to look at after the fact.

I wouldn't want to require "analyse_my_bench" to know how to pretty print basic benchmark output better than the library can already do.

> ```$ benchmark_my_stuff --benchmark_output=out.json --benchmark_output_format=json --benchmark_format=csv```
> it's very easy to get wrong. i mean, output is also to stdout, so output_format vs format is subtle.
> i'm wary of redundancy and confusion.

I'm sympathetic to that. A better way would be nice, or better names. However I don't think offering two output methods is too confusing. This behavior mirrors how LLVM's test runner LIT works. For example `$ lit -o out.json test.cpp` will give human readable output to stdout, and will print a copy of the JSON results to out.json.

In summary I agree with the general concern over confusion, but I think the feature is still useful to have.